Source/OrchBase/Composer.cpp

```diff
@@ -1059,25 +1665,71 @@ int Composer::generateSupervisor(ComposerGraphI_t* builderGraphI)
+    supervisor_cpp << "(static_cast<void*>(&(__reply.payload)));\n";
+    supervisor_cpp << "\tOS_PRAGMA_UNUSED(reply)\n\n";
+    
+    supervisor_cpp << "\tP_Pkt_t __bcast OS_ATTRIBUTE_UNUSED;\n";
+    supervisor_cpp << "\t" << outPktFmt << " broadcast OS_ATTRIBUTE_UNUSED= ";
+    supervisor_cpp << "static_cast<" << outPktFmt << ">";
+    supervisor_cpp << "(static_cast<void*>(&(__bcast.payload)));\n";
+    supervisor_cpp << "\tOS_PRAGMA_UNUSED(broadcast)\n\n";
+    
+    supervisor_cpp << "\tbool __rtsBcast = false;\n";
+    supervisor_cpp << "\tbool __rtsReply = false;\n\n";
+    
+    supervisor_cpp << supervisorOnImplicitHandler << "\n\n";
+    
+        // Reply packet logic
+    supervisor_cpp << "\tif(__rtsReply)\n\t{\n";
+    supervisor_cpp << "\t\tP_Addr_Pkt_t __oPkt;\n";
+    supervisor_cpp << "\t\tconst SupervisorDeviceInstance_t* tgt = ";
+    supervisor_cpp << "&DeviceVector[__inPkt->header.pinAddr];\n";
+    supervisor_cpp << "\t\t__reply.header.swAddr = tgt->SwAddr;\n";
+    supervisor_cpp << "\t\t__reply.header.swAddr |= ";
+    supervisor_cpp << "(P_CNC_IMPL << P_SW_OPCODE_SHIFT) & P_SW_OPCODE_MASK;\n";
+    supervisor_cpp << "\t\t__oPkt.hwAddr = tgt->HwAddr;\n";
+    supervisor_cpp << "\t\t__oPkt.packet = __reply;\n";
+    supervisor_cpp << "\t\t__outPkt.push_back(__oPkt);\n";
+    supervisor_cpp << "\t}\n\n";
+    
+        // Broadcast packet logic
+    supervisor_cpp << "\tif(__rtsBcast)\n\t{\n";
+    supervisor_cpp << "\t\t__bcast.header.swAddr = (P_ADDR_BROADCAST << ";
+    supervisor_cpp << "P_SW_DEVICE_SHIFT) & P_SW_DEVICE_MASK;\n";
+    supervisor_cpp << "\t\t__bcast.header.swAddr |= ";
+    supervisor_cpp << "(P_CNC_IMPL << P_SW_OPCODE_SHIFT) & P_SW_OPCODE_MASK;\n";
+    supervisor_cpp << "\t\tWALKCVECTOR(uint32_t,ThreadVector,threadTgt)\n\t{\n";
+    supervisor_cpp << "\t\t\tP_Addr_Pkt_t __oPkt;\n";
+    supervisor_cpp << "\t\t\t__oPkt.hwAddr = (*threadTgt);\n";
+    supervisor_cpp << "\t\t\t__oPkt.packet = __bcast;\n";
+    supervisor_cpp << "\t\t\t__outPkt.push_back(__oPkt);\n";
+    supervisor_cpp << "\t\t}\n";
+    
+    supervisor_cpp << "\t}\n\n";
+    
+    
+    supervisor_cpp << "\n\treturn 0;\n";
     supervisor_cpp << "}\n\n";
 
 
     // OnPkt - essentially a stub for now
-    supervisor_cpp << "int Supervisor::OnPkt(P_Pkt_t* inMsg)\n{\n";
+    supervisor_cpp << "int Supervisor::OnPkt(P_Pkt_t* __inPkt, ";
+    supervisor_cpp << "std::vector<P_Addr_Pkt_t>& __outPkt)\n{\n";
 
-    supervisor_cpp << "\t const SupervisorImplicitRecvMessage_t* message";
+    supervisor_cpp << "\tconst " << inPktFmt << " message";
     supervisor_cpp << " OS_ATTRIBUTE_UNUSED= ";
-    supervisor_cpp << "static_cast<const SupervisorImplicitRecvMessage_t*>(";
-    supervisor_cpp << "static_cast<const void*>(inMsg->payload));\n";
+    supervisor_cpp << "static_cast<const " << inPktFmt << ">(";
+    supervisor_cpp << "static_cast<const void*>(__inPkt->payload));\n";
     supervisor_cpp << "\tOS_PRAGMA_UNUSED(message)\n\n";
```

Comment:
As a note, this stuff is really hard to read. I know you know, and I know you know I know there's not enough time, but I thought I'd mention it for posterity. :fireworks:

Reply:
It is horrible, but short of templating there is not easy way to make it significantly easier to read.

I could split it up onto more lines, but that seems excessive.